cumulusci/tasks/bulkdata/utils.py

```diff
@@ -143,3 +159,17 @@ def check_for_row_error(self, result, row_id):
                 return self.row_error_count
             else:
                 raise BulkDataException(msg)
+
+
+def _consume(iterator):
+    # simplied from Python docs
+    collections.deque(iterator, maxlen=0)
+
+
+def _grouper(n, iterable):
+    it = iter(iterable)
+    while True:
+        chunk = tuple(itertools.islice(it, n))
+        if not chunk:
+            return
+        yield chunk
```

Comment:
We've already got one of these: get_batch_iterator in cumulusci.tasks.bulkdata.step

Reply:
Unified.